lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -220,7 +220,7 @@ def test_create_ccx(self):
 
         # assert ccx creator has role=ccx_coach
         role = CourseCcxCoachRole(course_key)
-        self.assertTrue(role.has_user(self.coach))
+        self.assertTrue(role.has_user(self.coach, refresh=True))
```

Comment:
It's calling out to a page above (self.client.get or whatever) and that page load creates a role.  So in the test context, `self.coach` has a stale set of roles.

So it's like...

- coach belongs to one role
- code is called that triggers a page load
-- page load adds new role to coach user, coach now belongs to two roles
- we check what roles the coach has, and since we don't reload before checking, it's still the same one role we saw at the beginning
- when we refresh, it picks up the new role, and all is well

Honestly, it's definitely simpler to reason about the roles if they're literally refreshed every single time you check if a user belongs to a role.. but the performance difference in tests is really stark.